### DISCUSSION TITLE
Custom searchForItems()

### DISCUSSION THREAD

**Discussion by ColeDCrawford** [algolia/instantsearch#6480] [Q&A]
I am trying to customize a `refinementList` (Vue InstantSearch). The `:searchable` attribute isn't working very well for me - I need to use a regex to search at any point in the facet option item string, not just from the front. But I can't find any examples of customizing the `searchForItems()` function for the default slot override. Are there any examples of how to go about this?

**Comment by Haroenv** [comment]
https://www.algolia.com/doc/guides/managing-results/optimize-search-results/override-search-engine-defaults/how-to/how-can-i-make-queries-within-the-middle-of-a-word/ would be an option for the regular search, not for search for facet values, but maybe with creativity a similar solution can be found.

What type of queries do you want to get results for? maybe other settings can help as well

**Comment by ColeDCrawford** [comment]
Hmm. My facet values look something like this:
```
          "pigment_paths": [
            "Clay white",
            "Arsenic sulfide",
            "Realgar",
            "Arsenic sulfide > Pararealgar",
            "Arsenic sulfide > Orpiment",
            "Indigo",
            "Vermilion or cinnabar",
            "Organic red",
            "Organic red > Insect red > Lac",
            "Carbon black",
            "Organic red > Insect red",
            "Ultramarine",
            "Unknown"
          ]
```

These are indexed as keywords (because they have to be for faceted search), but the lack of substrings means I can't search for a sublevel, e.g. "orpiment".

The data would be a good fit for the hierarchicalMenu but there's no searching at all for that component, and it doesn't seem possible to have multiple tree paths open at once, so due to those limitations I switched to rolling my own based on refinementList.

I have a tractable number of facets so a client-side search against the facet options is quite reasonable, if that's an option.

**Reply by Haroenv** [reply]
The solution I have in mind is the following then:

1. add another attribute which is only the "end" of the path, so in this case that would be `Orpiment` for example
2. depending on your search client version, intercept `search` (v5) or `searchForFacetValues` (v4) and translate the attribute into the other one that is just the leaf attribute you want to search into.

Alternatively you could indeed also use another method to search once you intercept that function, although that's a very broad topic of course.

What UI do you have for a searchable hierarchical menu? we've been thinking of it but could never find a suitable interface so didn't implement it for that reason.
